dali/c_api/c_api.cc

```diff
@@ -37,27 +37,23 @@ using dali::AccessOrder;
 using dali::CPUBackend;
 using dali::GPUBackend;
 
-namespace {
-
-bool dali_initialized = false;
-
-/**
- * Maps operator name to the batch size set prior to daliSetExternal... call.
- * Typically, this operator will be BatchSizeProvider.
- * Negative values denote max batch size (default state).
- * Typical usage:
- * auto *batch_size_map = reinterpret_cast<batch_size_map_t *>(handle->batch_size_map);
- */
 using batch_size_map_t = std::unordered_map<std::string /* op_name */, int /* batch_size */>;
+using data_id_map_t = std::unordered_map<std::string /* op_name */, std::string /* data_id */>;
 
 /**
- * Maps operator name to the data_id set prior to daliSetExternal... call.
- * Usually the operator with given `op_name` will be an InputOperator.
- * Typical usage:
- * auto *data_id_map = reinterpret_cast<data_id_map_t *>(handle->data_id_map);
+ * @brief Aggregates a DALI pipeline and auxiliary objects
  */
-using data_id_map_t = std::unordered_map<std::string /* op_name */, std::string /* data_id */>;
+struct DALIPipeline {
+  std::unique_ptr<dali::Pipeline> pipeline;
+  dali::Workspace workspace;
```

Comment:
Please follow our naming pattern `DaliPipeline`:
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

> For the purposes of the naming rules below, a "word" is anything that you would write in English without internal spaces. This includes abbreviations, such as acronyms and initialisms. For names written in mixed case (also sometimes referred to as "[camel case](https://en.wikipedia.org/wiki/Camel_case)" or "[Pascal case](https://en.wiktionary.org/wiki/Pascal_case)"), in which the first letter of each word is capitalized, prefer to capitalize abbreviations as single words, e.g., StartRpc() rather than StartRPC().


Additionally, how about not naming it `Pipeline`? Inside this structure there's more than the Pipeline - Workspace, weird maps, something that wraps CUDA stream... Maybe `DaliInstance` would be more suitable?

Reply:
In general: local conventions always prevail over global ones.
`[^A-Za-z0-9]Dali[A-Za-z0-9]+` - 149 occurrences
`[^A-Za-z0-9]DALI[A-Za-z0-9]+` - 1566 occurences

Reply:
I forgot the most important thing - this name is actually public and appears in the C API header as:
`typedef struct DALIPipeline *daliPipelineHandle_t`.
That's why it starts with DALI - like other structs/enums/... we use in C API. And that's why it's called a Pipeline - because that's what we call the handle.
In the implementation I could use a local typedef to make _our_ code cleaner, but wouldn't that be an overkill?

Reply:
Still, there is a clear and concise rule in our style guide, how we should format acronyms. I'm aware, that it's not consistent in our codebase, but since the name is public, I'd vote for keeping the rule - we're publicly stating that it's our rule after all.

Should you believe that we should break this rule, please elevate this issue to the team meeting, like we tend to do in such cases.

Reply:
I agree that `DALI*` appears way more often in the search query. Still, this is not something we use in this file, and introducing it here breaks this file's consistency as well.
If we really want to follow the convention we keep in this file it should be rather `daliPipeline`.

Reply:
It is not about bikeshedding but about understanding the other's point of view and ideas that don't deserve to be bashed.
In most cases, people are fine accepting the suggestions from the reviewers if it is about my opinion against yours (and that is the case I believe). This is approach is encouraging the reviewers to add any comment they feel can improve the code. The other way around is very discouraging and long term affects the code quality.